camel/memories/agent_memories.py

```diff
@@ -104,21 +108,68 @@ def get_context_creator(self) -> BaseContextCreator:
         return self._context_creator
 
 
+class GraphDBMemory(AgentMemory):
+    r"""An agent memory wrapper of `GraphDBBlock`. This memory interacts with
+    a graph database to store and retrieve knowledge graphs.
+
+    Args:
+        context_creator (BaseContextCreator): A model context creator.
+        storage (BaseGraphStorage, optional): A graph storage backend. If
+            `None`, a default implementation will be used. (default: `None`)
+        query (str, optional):
+            The query to retrieve data from the graph database.
+    """
+
+    def __init__(
+        self,
+        context_creator: BaseContextCreator,
+        storage: Optional[BaseGraphStorage] = None,
+        query: str = "",
+    ) -> None:
+        self._context_creator = context_creator
+        self._graphdb_block = GraphDBBlock(storage=storage)
+        self._query = query
+
+    def retrieve(self) -> List[ContextRecord]:
+        return self._graphdb_block.retrieve(self._query)
+
+    def write_triplet(self, subj: str, obj: str, rel: str) -> None:
+        r"""Writes a triplet to the graph database.
+
+        Args:
+            subj (str): The subject of the triplet.
+            obj (str): The object of the triplet.
+            rel (str): The relationship between subject and object.
+        """
+        self._graphdb_block.write_triplet(subj, obj, rel)
```

Comment:
I think here should be `write_records`  using `self._graphdb_block.write_record()`